impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Abstract.js

```diff
@@ -256,10 +271,94 @@ define([
 
       this._configureOptions();
 
+      this._chart.on("dblclick", function (params) {
+        this._onExecute(params);
+      }, this);
+
+      // When 'click' and 'dblclick' are bound to same element
+      // and user double-clicks, 'click' is fired twice followed by dbl-click
+      // Reference: https://stackoverflow.com/a/7119911
+      var clickTimeout;
+      this._chart.on("click", function (params) {
+        clearTimeout(clickTimeout);
+        var me = this;
+        clickTimeout = setTimeout(function() {
+          me._onSelect(params);
+        }, 300);
+      }, this);
+
       // Draw the chart
       this._chart.setOption(this._echartOptions);
     },
 
+    _onExecute: function(params) {
+      params.cancelBubble = true;
+      var model = this.model;
+      if(model.isDirty) {
+        return;
+      }
+
+      var filter = null;
+      var srcEvent = params.event;
+
+      var filter = this._createFilter(params);
+
+      if(filter === null) {
+        return;
+      }
+
+      model.execute({
+        dataFilter: filter,
+        position: srcEvent ? {x: srcEvent.offsetX, y: srcEvent.offsetY} : null
+      });
+    },
+
+    _onSelect: function(params) {
+      params.cancelBubble = true;
+      var model = this.model;
+      if(model.isDirty) {
+        return;
+      }
+
+      var filter = null;
+      var srcEvent = params.event.event;
+
+      var filter = this._createFilter(params);
+
+      if(filter === null) {
+        return;
+      }
+
+      model.select({
+        dataFilter: filter,
+        selectionMode: srcEvent.ctrlKey || srcEvent.metaKey ? "toggle" : "replace"
+      });
+    },
+
+    _createFilter: function(params) {
+      var model = this.model;
+
+      var varsMap = this._getVars(params.data.visualKey);
+
+      var keyDataCellsMap = sceneUtil.invertVars(varsMap, model);
+
+      return dataUtil.createFilterFromCellsMap(keyDataCellsMap, model.data);
+    },
+
+    _getVars: function(eventData) {
```

Comment:
Couldn't `eventData` more clearly, and equally, be called `visualKey`?

Reply:
Will make the change.